renderdoc/android/android.cpp

```diff
@@ -337,6 +343,49 @@ AndroidVersionCheckResult CheckAndroidServerVersion(const rdcstr &deviceID, ABI
   return AndroidVersionCheckResult::WrongVersion;
 }
 
+Process::ProcessResult InstallAPK(const rdcstr &deviceID, const rdcstr &apk, int apiVersion)
+{
+  Process::ProcessResult adbInstall;
+  if(apiVersion >= 30)
+  {
+    adbInstall = adbExecCommand(deviceID, "install -r -g --force-queryable \"" + apk + "\"");
+  }
+  else
+  {
+    adbInstall = adbExecCommand(deviceID, "install -r -g \"" + apk + "\"");
+  }
+
+  return adbInstall;
+}
+
+AndroidInstallPermissionCheckResult CheckAndroidServerInstallPermissions(const rdcstr &deviceID,
+                                                                         rdcstr packageName,
+                                                                         int apiVersion)
+{
+  // Permissions check only applicable to API > 30
+  if(apiVersion < 30)
+  {
+    return AndroidInstallPermissionCheckResult::Correct;
+  }
+  // Command to check that the Android Server is queryable by other APKs (API>30)
+  // for RenderDoc layer discovery.
+  Process::ProcessResult adbCheck =
+      adbExecCommand(deviceID,
+                     "shell \"dumpsys package queries | sed -n '/forceQueryable/,/queries via "
+                     "package name/p' | grep " +
```

Comment:
This is quite scary especially for something that will end up running on all devices. adb commands are a large potential point of failure and should be minimised as much as possible. Running complex commands on the shell like this runs the risk of failures and/or hangs.

Instead I think this should be just running `dumpsys package` and then doing the parsing on the output as that will be significantly safer.

Reply:
Good point, I have made this adjustment, bearing in mind the output of `dumpsys package queries` may be quite long (1.5k lines on a test device).